src/levanter/infra/ray_tpu.py

```diff
@@ -448,92 +384,27 @@ def _handle_ray_error(tpu_info: _TpuInfo, e: RayError):
-    * Ray runs tasks in a long-running daemon, so the lockfile persists across tasks.
-    * jax.distributed likes to only be called once per process, even if you call shutdown
-
-    """
-    if isinstance(remote_fn, RemoteFunction):
-        fn = remote_fn._function
-
-        @functools.wraps(fn)
-        def wrapped_fn(*args, **kwargs):
-            return _separate_process_fn(fn, args, kwargs)
-
-        # We need these arguments to be able to reconstruct the remote function
-        # def __init__(
-        #         self,
-        #         language,
-        #         function,
-        #         function_descriptor,
-        #         task_options,
-        # ):
-        remote_fn = RemoteFunction(
-            language=remote_fn._language,
-            function=wrapped_fn,
-            function_descriptor=remote_fn._function_descriptor,
-            task_options=remote_fn._default_options,
-        )
-        return remote_fn
-    else:
-        return functools.partial(_separate_process_fn, remote_fn)
-
-
-def _separate_process_fn(underlying_function, args, kwargs):
-    """
-    Helper function for _forkify_remote_fn. This runs the function in a separate process.
-    """
-
-    def target_fn(queue, args, kwargs):
-        try:
-            # Call the original function
-            result = underlying_function(*args, **kwargs)
-            queue.put((True, result))  # Success, put the result
-        except Exception as e:
-            # Capture and return the full traceback in case of an exception
-            info = ser_exc_info(e)
-            queue.put((False, info))
-
-    queue = multiprocessing.Queue()
-    process = multiprocessing.Process(target=target_fn, args=(queue, args, kwargs))
-    process.start()
-    process.join()
-
-    # Retrieve the result or error from the queue
-    success, value = queue.get()
-
-    if success:
-        return value
-    else:
-        value.reraise()
-
-
-def _hacky_remove_tpu_lockfile():
-    """
-    This is a hack to remove the lockfile that TPU pods create on the host filesystem.
-
-    libtpu only allows one process to access the TPU at a time, and it uses a lockfile to enforce this.
-    Ordinarily a lockfile would be removed when the process exits, but in the case of Ray, the process is
-    a long-running daemon that doesn't typically exit until the node is shut down. This means that the lockfile
-    persists across Ray tasks. This doesn't apply to our docker-based workloads, but it does apply to other
-    tasks that use JAX directly.
-    """
-    try:
-        os.unlink("/tmp/libtpu_lockfile")
-    except FileNotFoundError:
-        pass
-    except PermissionError:
-        logger.warning("Failed to remove lockfile")
-        try:
-            os.system("sudo rm /tmp/libtpu_lockfile")
-        except Exception:  # noqa
-            pass
+# TODO(cathy): this is unused, do we need it?
```

Comment:
oh yeah i don't think we need it anymore. I had it before we started forking a training process